canvas/gradient.go

```diff
@@ -3,49 +3,101 @@ package canvas
 import (
 	"image"
 	"image/color"
-	"image/draw"
 	"math"
-)
 
-const (
-	// GradientDirectionVertical defines a vertically traveling gradient
-	GradientDirectionVertical GradientDirection = iota
-	// GradientDirectionHorizontal defines a horizontally traveling gradient
-	GradientDirectionHorizontal
-	// GradientDirectionCircular defines a gradient traveling radially from a center point outward
-	GradientDirectionCircular
+	"fyne.io/fyne"
 )
 
-// GradientDirection defines a starting or stopping location
-type GradientDirection int
+// Gradient describes a gradient fade between two colors.
+type Gradient interface {
+	fyne.CanvasObject
+	calculateGradient(x, y, w, h float64) float64
+	Compute(w, h int) image.Image
```

Comment:
This moved from Generator before and the comment about not using directly - has the intent of the method changed?
As I understood it this could be moving into the driver for optimisation purposes and so people were warned in the docs. Do you think this has changed now?

Reply:
`Generator` returns a `func` which is okay for `canvas.Raster` which does not know anything about its generation (the function is handed in on creation of the `Raster`). But for the gradients I can't see any advantage in the indirection of returning a generator method instead of just perform the computation.